Code/Tools/ProjectManager/Source/GemCatalog/GemCatalogHeaderWidget.cpp

```diff
@@ -288,29 +289,40 @@ namespace O3DE::ProjectManager
         }
     }
 
-    void CartOverlayWidget::GemDownloadProgress(const QString& gemName, int percentage)
+    void CartOverlayWidget::GemDownloadProgress(const QString& gemName, int bytesDownloaded, int totalBytes)
     {
         QWidget* gemToUpdate = m_downloadingListWidget->findChild<QWidget*>(gemName);
         if (gemToUpdate)
         {
             QLabel* progressLabel = gemToUpdate->findChild<QLabel*>("DownloadProgressLabel");
-            if (progressLabel)
+            QProgressBar* progressBar = gemToUpdate->findChild<QProgressBar*>("DownloadProgressBar");
+
+            if (totalBytes != 0)
             {
-                progressLabel->setText(QString("%1%").arg(percentage));
+                int downloadPercentage = static_cast<int>((bytesDownloaded / static_cast<float>(totalBytes)) * 100);
+                if (progressLabel)
+                {
+                    progressLabel->setText(QString("%1%").arg(downloadPercentage));
+                }
+                if (progressBar)
+                {
+                    progressBar->setValue(downloadPercentage);
+                }
             }
-            QProgressBar* progressBar = gemToUpdate->findChild<QProgressBar*>("DownloadProgressBar");
-            if (progressBar)
+            else
             {
-                progressBar->setValue(percentage);
+                if (progressLabel)
+                {
+                    progressLabel->setText(QLocale::system().formattedDataSize(bytesDownloaded));
+                }
+                if (progressBar)
+                {
+                    progressBar->setRange(0, 0);
```

Comment:
The progress bar looks kinda weird in the screenshot you posted. Is it because of this? If so I would just hide it. And make sure to show it again when you should if you do hide it.

Reply:
It would have looked better as a gif :) It's the animated scrolling state of a progress bar you generally see when progress is unknown.